pkg/common/cns-lib/vsphere/hostsystem.go

```diff
@@ -219,3 +219,41 @@ func GetSharedDatastoresForHosts(ctx context.Context, hosts []*HostSystem) ([]*D
 	}
 	return sharedDatastores, nil
 }
+
+// GetAllAccessibleDatastoresForHosts returns datastores accessible to hosts mentioned in the input parameter.
+func GetAllAccessibleDatastoresForHosts(ctx context.Context, hosts []*HostSystem) ([]*DatastoreInfo, error) {
+	log := logger.GetLogger(ctx)
+	var allAccessibleDatastores []*DatastoreInfo
+
+	for _, host := range hosts {
+		accessibleDatastores, err := host.GetAllAccessibleDatastores(ctx)
+		if err != nil {
+			log.Warnf("failed to fetch datastores from host %+v. Error: %+v",
+				host, err)
+			continue
+		}
+		if len(accessibleDatastores) == 0 {
+			return nil, logger.LogNewErrorf(log, "failed to find accessible datastores for host %+v.",
```

Comment:
Does it make sense to error out here? Or do we move on to other hosts?

Reply:
Based on discussion I had with Divyen, we concluded to have the error returned here in cases where we have permission issue and driver is unable to find datastore on the host.